### ISSUE TITLE
Different output across multiple runs for same input and decode parameters 

### ISSUE THREAD

**Issue by udaypk** [google-research/text-to-text-transfer-transformer#313]
We trained a model for a summarization like task. But when we run an inference with MtfModel.predict function with beam_size=1, and temperature=1.0, we are getting different outputs for the same input text. I'm not sure how to set the random seed to a constant so that we get same output for same input across runs. 

I found the following Pytorch code https://colab.research.google.com/github/enzoampil/t5-intro/blob/master/t5_qa_training_pytorch_span_extraction.ipynb#scrollTo=HM4stMDvJYLt&line=23&uniqifier=1

Can anyone suggest how to set the random seed in native TF.

**Comment by craffel** [comment]
I'm not 100% sure but it should be enough to do
```Python
tf.random.set_random_seed(0)
random.seed(12345)
```
and also set t5.preprocessors.num_parallel_calls.deterministic = True via gin if you are using a stochastic preprocessor.

**Comment by udaypk** [comment]
> I'm not 100% sure but it should be enough to do
> 
> ```python
> tf.random.set_random_seed(0)
> random.seed(12345)
> ```
> 
> and also set t5.preprocessors.num_parallel_calls.deterministic = True via gin if you are using a stochastic preprocessor.

thanks for the suggestion but this is not working. I'm still getting random output.

**Comment by adarob** [comment]
@nshazeer do you know if this is possible?

**Comment by jspablo** [comment]
Can we reopen this issue? Using the official notebook [t5-trivia.ipynb](https://github.com/google-research/text-to-text-transfer-transformer/blob/main/notebooks/t5-trivia.ipynb) setting the seed before loading the model and the predict cell is not working.

I cannot find how to set `t5.preprocessors.num_parallel_calls.deterministic = True` via gin from the notebook. Can you give more detail about this?

It would be great to keep this deterministic behaviour when deploying the model to Tensorflow Serving
